packages/website/src/components/playground/PlaygroundRoot.tsx

```diff
@@ -0,0 +1,201 @@
+import { useWindowSize } from '@docusaurus/theme-common';
+import type * as ESQuery from 'esquery';
+import React, { useEffect, useRef, useState } from 'react';
+import type { ImperativePanelHandle } from 'react-resizable-panels';
+import { Panel, PanelGroup, PanelResizeHandle } from 'react-resizable-panels';
+
+import ASTViewer from '../ast/ASTViewer';
+import ConfigEslint from '../config/ConfigEslint';
+import ConfigTypeScript from '../config/ConfigTypeScript';
+import LoadingEditor from '../editor/LoadingEditor';
+import useHashState from '../hooks/useHashState';
+import EditorTabs from '../layout/EditorTabs';
+import { createFileSystem } from '../linter/bridge';
+import type { UpdateModel } from '../linter/types';
+import { isCodeFile } from '../linter/utils';
+import { TypesDetails } from '../typeDetails/TypesDetails';
+import { debounce } from '../util/debounce';
+import { defaultConfig, detailTabs } from './config';
+import { ErrorsViewer } from './ErrorsViewer';
+import { ESQueryFilter } from './ESQueryFilter';
+import OptionsSelector from './OptionsSelector';
+import styles from './playground.module.css';
+import type { ErrorGroup, PlaygroundSystem } from './types';
+
+function PlaygroundRoot(): JSX.Element {
+  const [config, setConfig] = useHashState(defaultConfig);
+
+  const [system] = useState<PlaygroundSystem>(() => createFileSystem(config));
+  const [activeFile, setFileName] = useState(`file.${config.fileType}`);
+  const [editorFile, setEditorFile] = useState(`file.${config.fileType}`);
+
+  const [errors, setErrors] = useState<ErrorGroup[]>([]);
+  const [astModel, setAstModel] = useState<UpdateModel>();
+  const [esQueryFilter, setEsQueryFilter] = useState<ESQuery.Selector>();
+  const [showModal, setShowModal] = useState<string | false>(false);
```

Comment:
Tangentially related to this PR, but I do wish the modal didn't take over the full playground. It'd be nice to limit it to just the editor tab it's relevant for. No need to block the options or AST/errors panes...

Reply:
agree, but this is oo scope for this change